### ISSUE TITLE
Cross-platform: Untranslatable words

### ISSUE THREAD

**Issue by turt2live** [vector-im/element-web#17112]
We have a bunch of technical terminology in our app, and while we should make an effort to tone it down, we also have a responsibility to have accurate translations of the words we use.

Many words, like Spaces, might not have direct translations and instead be approximate based on various criteria. We should set up a [Weblate Glossary](https://docs.weblate.org/en/latest/user/glossary.html) with all the Element/Matrix specific terminology to give translators the best chance at translating them.

**This issue exists to collect those words, and the languages that have particular difficulty translating them.**

**Comment by turt2live** [comment]
Spaces is one such word, as it doesn't translate very nicely to other languages. Possible alternatives for translators to consider would be "Workspaces" or "Communities".

**Comment by libexus** [comment]
For the german translation, we currently just use "Spaces" without translating it because it has gotten a meaning like [Dandellion (nb_NO 🇳🇴) said](https://matrix.to/#/!yvhoprZiKmhjxKfNHa:matrix.org/$xeZiKHQDe_7FpjI3Y0O7D9FSPUCke_d0TpfmoPRF7ok). I think that's not a problem since we use *a lot* of english terms in Germany+Austria already. But I think the other translators aren't to happy about it tough(?)

But do you think we should change them later on (in case we found a proper german word) or keep it?

(CC @a000a @Sven792)

**Comment by insidan** [comment]
Hi, I'm user vintergatan on translate.element.io (I'm also completely new to Github, not sure how things work here)

I have translated parts of Element Web and iOS to my native language Swedish. I want to mention **end-to-end encryption** since it's a clearly defined technical term (I think?) but in my language lacks a clear translation.

tl;dr: I would like to keep the English e2e in Swedish strings and not invent a new term

I have made suggestions on Weblate to keep the English term in the Swedish translation, _end-to-end-kryptering_. But some other translator suggested the translation _totalsträckskryptering_ which seems like a very literal translation that I don't think is a good choice. But I don't want to participate on an edit war so I have not made new suggestions on Weblate.

I'm gonna throw in some links showing how major services translate e2e encryption in Swedish. I'm a strong believer when it comes to crowd sourced translations to not try and reinvent the word wheel. 

Apple: Keeps the English term
https://support.apple.com/sv-se/HT202303

WhatsApp: Keeps the English term in some info web pages but uses _Komplett kryptering_ ("Complete encryption") in the app
https://www.whatsapp.com/features/?lang=sv

Google Duo: Keeps the English term
https://support.google.com/duo/answer/9280240?hl=sv

Facebook Messenger: uses _krypterade under hela processen_ which is not a term, just a description ("encrypted during the whole process")

**Comment by Smarre** [comment]
Hi,

First of all, thank you for this initiative. While actual translation can be said to be easy, localization is hard: the translated content needs to be fit to the user in same way the original text feels natural to a British person. Since all the languages are different, and especially between language groups the conventions and whole structure of the language can be wildly different, localization naturally means adaptation to the local environment.

My native language, Finnish, has most of the problems with translation with different terms. Finnish is heavily conjugated language, meaning the words changes their form depending on how they are used in sentences. For example, ”I am cat” would literally be something like ”minä olla kissa”, which is wrong. Instead we’re using ”olen kissa”. ”I have a cat” would be ”minulla on kissa”.

This means that there is often much fitting to be done with terms and such. Especially things like input parameters, which subtitutes for example the ”I” to above examples, would be very hard to translate. We basically need to reword all of those, and sometimes need to resort to somewhat clunky wording. This is unavoidable since it would be very time consuming to try to map everything properly on code side. 

From this we arrive to the problem with the term ”Spaces”. The word actually has quite many translations. Here probably closest one is ”avaruus”, which means the things outside the earth, but also more generally, the space around something, in an abstract way. This has no resemblance with the original word, which is bad in terms of marketing, but also would not fit to Finnish language.

Sometimes English terms that gets adopted to Finnish results to very unfortunate situation. For example ”app” in a sentence ”I made an app” would be translated either ”tein apin” or ”tein appin”. Former would mean that I created an API and latter would not fit to Finnish at all (and in practice everyone is saying first). This again has no resemblance to the original word. To add to the injury, Finnish often has vocal endings for words, so ”app” gets often used as ”appi”, which means your partner’s father. So depending on the context, it’s starting to get pretty vague.

Similar problems would be present with ”space”, as it does not fit to Finnish well. Something like ”Matrix Wallet” (or Element :) for example would fit more easily, since it would have unambiguous pronounciation and spelling (spelling and pronounciation is really almost exactly same for Finnish). To illustrate, space could be pronounced as ”speissi” or ”spase” here. Latter is very bad. Former is what people would be using when speaking, rather than writing. And sometimes in writing too, I suppose. 

Due to conjugation, ”Spaces” as a pseudo-singular is much worse. Even if it would mean plural in English, it would be very bad in terms of gramar to use it as plural in Finnish. Finnish plural for ”Space” would be ”Spacet”. Compare to ”Avaruus”, ”Avaruudet”.  Since plural ”Spaces” would not mean anything in Finnish, all kind of fishy conventions would be necessary to be done in order to work around all of this. Most like something like ”Space-kokonaisuudet” (collections of Space), ”Space-avaruudet” (spaces of Space) or something similar would be needed to be used in order to better fit to Finnish. Actual used circumvention is context dependent.

That should be enough about Spaces. This of course concerns other terms too. As mentioned above, for example the name Element does not suffer of this problem, since it fits to Finnish much easier. That is, it is easy to use as an explicit name, where ”Spaces” would nigh impossible, and ”Space” would be awkward. latter is workable though, if it is a must. I’d prefer not to. For other terms, I don’t know if there is anything that is not translated, outside of the Element name. 

..

Someone raised the topic of E2EE and its myriad forms. I agree with that is fairly technical terms, but it seems to be about everywhere. In few place I’ve just replaced it with translation of ”encryption” or so, but mostly it is a complete mess at the moment.

I originally translated it with slightly inaccurate ”osapuolten välinen salaus”, which literally means ”encryption between participants”. Someone went and reverted this to an awkward ”päästä päähän -salaus”, which is more literal translation of E2EE. the ”-salaus” means that ”päästä päähän” is a proper noun, i.e. name of the feature, rather than just all words being common nouns. this means that this is very annoying to conjugate, and IMHO this fits to Finnish very badly. I’ve been getting complaints about this ”päästä päähän -salaus” versions being confusing, so I’ll probably just revert back to my original version when I get to it. But for us the best solution probably would be use some simpler term, like ”encryption”, in most of the strings. 

..

There is many other inconsistencies in our translation also, and I think major problem that has contributed to it is that there is no common dictionary. Currently we basically would need to copy terms from Web dictionary to Android dictionary, and it seems that no-one has done it, so terms are sometimes wildly different. 

There was a speak about dictionary for these terms that needs to be unified, but I think the dictionary in Weblate, that is language-specific, is different thing. For me, instructions about how to translate a term is more of a document somewhere, or context written to the string. Not that there really is context for most of the strings at the moment... Context is always good, because it allows translator to know if something else than literal translation is good. And at least for us, often there is many different translations for a word. The [Space](https://fi.wiktionary.org/wiki/space) is a good example.

So, one language-specific dictionary for all Matrix projects instead of per-project dictionaries, please.

**Comment by MurzNN** [comment]
Is there a separate task (issue) for "should set up a Weblate Glossary with all the Element/Matrix specific terminology"? I created before the https://github.com/vector-im/element-android/issues/1903 issue bit it closed now in favor of this.

**Comment by jelv** [comment]
For Dutch we use the [glossary](https://translate.element.io/glossaries/element-web/nl/) a lot and translated everything except Spaces. Some pain points I've seen are:

Matrix terms:
- **Rooms:**
We use the Dutch "conversations" and "group conversation" for the combination of: rooms, chats, groups, group chat, one-to-one, DM's, direct message, direct chat, private chat and conversations. Not sure if we should coordinate such translation between languages?
- Homeserver
- User ID or Matrix ID
- Spaces
 

Technical terms like:

- Secure Backup
- Security Key
- Security Phrase
- Encryption key
- Cross-signing
- End-to-end encryption
- Passphrase
- Peer-to-peer
- Single-sign-on
- Verify
